frontend/src/App.svelte

```diff
@@ -3,12 +3,17 @@
   import { Link } from 'carbon-components-svelte'
   import { routes } from './routes'
   import LanguageSelector from './lib/LanguageSelector.svelte'
+  import { setLanguage } from './lib/stores'
+
+  function routeLoading(event) {
+    setLanguage(event.detail.params.lang)
+  }
 </script>
 
 <div class="container">
   <LanguageSelector />
   <div class="content">
-    <Router {routes} />
+    <Router {routes} on:routeLoading={routeLoading} />
```

Comment:
This is not great, you should use the parameter from routes way of doing this, so add `export let params` to your `Main` component

https://github.com/ItalyPaleAle/svelte-spa-router#parameters-from-routes